docs/adrs/003-bugzilla-response-codes.md

```diff
@@ -0,0 +1,185 @@
+- Keeping all services within the container make end-to-end testing and debugging easier (compared to option 4)
+
+**Cons:**
+- Increases complexity of the containers
+- Data will not persist container restarts
+- High effort for engineers to build and maintain
+- Less intuitive to users and engineers, we would need to report on cache/queue metrics to logs
+- Data could be processed out of order, causing newer updates to get lost
+
+### Option 4: Use a simple DLQ (dead letter queue)
+We would always return 200, but any events that fail to process internally would get sent to a DLQ and be replayed later if needed. This could be a storage bucket, kubernetes volume, or table in a database. A scheduled kubernetes job would then run to try and pick these up and reprocess them later (every 4 hours, for example). 
+
+After too many failed attempts the payload would be marked as unprocessable (setting a flag in the table, or updating the file name).
+
+**Decision Drivers**
+- Amount of initial engineering effort: mid, minimal added infrastructure (database/table or k8s volume or storage bucket)
+- Amount of maintenance effort: mid-low (mid if we spin up a new database)
+- Overall performance of JBI: high
+- How intuitive the solution: high to engineers, low to users
+
+**Pros:**
+- Durable and expandable solution
+- Does not reduce JBI throughput
+- Intuitive to engineers
+
+**Cons:**
+- Added infrastructure
+- Not intuitive to end users unless we build additional reporting so they know why an update didn't come over
+- Data could be processed out of order, causing newer updates to get lost
+
+### Option 5: Use a dedicated queue solution
+We would have a dedicated service that accepts all API calls from Bugzilla and puts them into a queue (apache kafka, rabbitMQ, etc). JBI would shift to being a downstream service and process these events asynchronously. Any events that fail to process would get sent to a DLQ (dead letter queue) that could be replayed later if needed.
+
+There are plenty of existing solutions we could use to solve this problem from a technical perspective. A separate ADR would be done to identify the best answer if we choose to go this route.
+
+**Decision Drivers**
+- Amount of initial engineering effort: high, building out more infrastructure
+- Amount of maintenance effort: high, maintaining more infrastructure
+- Overall performance of JBI: highest, event driven
+- How intuitive the solution: high - we'll have reporting on queue metrics
+
+**Pros:**
+- Most durable solution
+- Does not reduce JBI throughput
+- Intuitive to users and engineers, we can see and report on data in queue
+
+**Cons:**
+- Most complex solution
+- Highest effort for engineers to build and maintain
+
+### Option 6: A combination of the above
+Example: We could create a simple DLQ (a table in postgres) external queue for re-processing and then alert users if the DLQ grows too quickly.
+
+## Decision Outcome
+
+Pending discussion
```

Comment:
Isn't this exposed in `## Proposed Solution`?

Reply:
adjusting that, hopefully we'll finalize our approach tomorrow